File: ios/Arculus/CSDK/WalletResetWallet.swift
import CSDK

final class WalletResetWallet: CSDKAPICall<Void> {
    override func request() async throws -> [Data] {
        var len: size_t = 0
        
        guard let bytes = WalletResetWalletRequest(wallet, &len) else {
            throw CSDKAPICallError(.requestCreationFailed("reset wallet"))
        }
        
        return [Data(bytes: bytes, count: len)]
    }
    
    override func response(bytes: [UInt8]) async throws -> ResponseType {
        let result = WalletResetWalletResponse(wallet, bytes, bytes.count)
        
        try CSDKAPICallError.validateWalletResponseCode(rc: result)
    }
}

File: android/src/main/java/com/aptoslabs/arculus/csdk/WalletResetWallet.kt
package com.aptoslabs.arculus.csdk

import com.sun.jna.Pointer

class WalletResetWallet(wallet: Pointer) : CSDKAPICall<Unit>(wallet) {
  override suspend fun request(): Array<ByteArray?> {
    val len = SizeTByReference()

    val pointer = CSDK.WalletResetWalletRequest(wallet, len)

    if (pointer == Pointer.NULL) throw CSDKAPICallRequestCreationException()

    return arrayOf(pointer.getByteArray(0, len.toIntChecked()))
  }

  override suspend fun response(bytes: ByteArray) {
    val rc = CSDK.WalletResetWalletResponse(wallet, bytes, bytes.size)

    validateWalletResponse(rc)
  }
}
